### DISCUSSION TITLE
Pushing prompt to the top line without clearing current output

### DISCUSSION THREAD

**Discussion by aksh1618** [kovidgoyal/kitty#3721] [Q&A] [answered]
Hi! I'm exploring and setting up kitty to match my workflow. The features (speed, configuration, kittens, so many awesome things!), ideology, community and how active @kovidgoyal is, made me want to switch to Kitty (along with a broken progress bar rendering in my existing terminal, ofc), and I'm loving it so far! 

I just have a couple things left to figure out, that I'm used to in my current workflow, one of them is being able to push the current visible output up so that the prompt comes to the top line. Clear (Ctrl + L) does this but also clears some output from the hostory. Is there a way to do this in kitty?

**Comment by kovidgoyal** [comment, marked as answer]
https://sw.kovidgoyal.net/kitty/conf.html#shortcut-kitty.Reset-the-terminal

**Reply by aksh1618** [reply]
Kitty just keeps exceeding my expectations, the consideration for user needs and experience is unbelievable. I was expecting some option to clear it this way, not support for probably every possible way of resetting, this is so amazing! And on top of that I go in the configuration file, and *exactly* what I need (remapping ctrl+ l to do this) is already there, just waiting to be uncommented. Your attitude and dedication is inspiring! :heart:

**Reply by kovidgoyal** [reply]
Note that the remapping for ctrl+l must be done in the shell, not in kitty.

**Reply by aksh1618** [reply]
Thanks, found an example also in the config, after updating kitty to the latest version, for zsh:

```zsh
# Fix Ctrl + L (From kitty config comments)'
scroll-and-clear-screen() {
    printf '\n%.0s' {1..$LINES}
    zle clear-screen
}
zle -N scroll-and-clear-screen
bindkey '^l' scroll-and-clear-screen
```

But had to rename the config to get the latest one get generated. Is there a recommended way to get the latest config after upgrading kitty to the latest version, or is this the best way?
